solr/core/src/java/org/apache/solr/response/TextResponseWriter.java

```diff
@@ -55,6 +58,10 @@
 
   protected Calendar cal;  // reusable calendar instance
 
+  private static final ReturnFields DUMMY_RETURN_FIELDS = new SolrReturnFields();
```

Comment:
Yeah; `== null` was indeed an obvious first choice; but that doesn't work because it's being used as shorthand to check whether we're in the appropriate state to do special handling. And some of the "inappropriate" cases are signified by `null` explicitly passed, which would have matched on `null`. I could at least comment it or name the variable more clearly. Good point.

Reply:
Awesome... thanks!